ratis-server/src/main/java/org/apache/ratis/server/impl/StateMachineUpdater.java

```diff
@@ -190,11 +189,11 @@ public void run() {
           reload();
         }
 
-        final MemoizedSupplier<List<CompletableFuture<Message>>> futures = applyLog();
-        checkAndTakeSnapshot(futures);
+        applyLogFutures = applyLog(applyLogFutures);
+        checkAndTakeSnapshot(applyLogFutures);
 
         if (shouldStop()) {
-          checkAndTakeSnapshot(futures);
+          checkAndTakeSnapshot(applyLogFutures);
```

Comment:
Replace it with `applyLogFutures.get()` since 
1. checkAndTakeSnapshot(..) is already invoked above, and
2. checkAndTakeSnapshot(..) does not call applyLogFutures.get() when shouldTakeSnapshot() is false.

Reply:
yeah you are right we don't need this take snapshot

Reply:
done